inf-clojure.el

```diff
@@ -511,10 +519,11 @@ Fallback to `default-directory.' if not within a project."
 
 (defun inf-clojure-project-type ()
   "Determine the type, either leiningen or boot of the current project."
-  (let ((default-directory (inf-clojure-project-root)))
-    (cond ((file-exists-p "project.clj") "lein")
-          ((file-exists-p "build.boot") "boot")
-          (t nil))))
+  (when (null inf-clojure-project-type)
```

Comment:
Why don't you use `or` here?

Reply:
will do

Reply:
No wait, the code should run if this is `nil`. So basically something else than `nil` will not trigger the detection.

Reply:
Well, that's what `or` will do - `(or type (detect-type))`. That seems pretty clear to me.

Reply:
You are right, I was confused for a moment :smile: